api/api.go

```diff
@@ -0,0 +1,89 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/NebulousLabs/skynet-accounts/database"
+
+	"github.com/NebulousLabs/skynet-accounts/build"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+const (
+	// DefaultTimeoutDB defines the longest a DB operation can take before
+	// triggering a timeout. In seconds.
+	DefaultTimeoutDB = 10
+	// DefaultTimeoutRequest defines the longest an API request can take before
+	// triggering a timeout. In seconds.
+	DefaultTimeoutRequest = 30
+)
+
+// API is ...
+type API struct {
+	staticDB     *database.DB
+	staticRouter *httprouter.Router
+}
+
+// New returns a new initialised API.
+func New() (*API, error) {
+	router := httprouter.New()
+	router.RedirectTrailingSlash = true
+
+	api := &API{
+		staticRouter: router,
+	}
+	api.buildHTTPRoutes()
+
+	ctx, cancel := context.WithTimeout(context.Background(), DefaultTimeoutDB)
+	defer cancel()
+	db, err := database.New(ctx)
```

Comment:
The database should be passed in. It should probably also be an interface that's passed in. That way it's more modular in case we ever swap out the database in the future. It also allows for easier mocking.

Reply:
F/U: https://github.com/NebulousLabs/skynet-accounts/issues/5